sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -113,6 +113,11 @@ object NestedColumnAliasing {
     case _: Sample => true
     case _: RepartitionByExpression => true
     case _: Join => true
+    case x: Filter => x.child match {
+      case _: Window => true
```

Comment:
Why we need this? We cannot support the window case only with the change `case _: Window => true`?

Reply:
Looks like the plan is a `Project -> Filter -> Window`. If we only do `case _: Window => true`, the projection aliasing won't be available at the `Window` stage, and can't be passed onto later stages described in the ticket.

Reply:
But, `Project->Filter->Sample` has the same issue. So, it seems to be a different issue.

Reply:
Then just add `case _: Filter => true`, if you want to let project pushed through `Filter`?

Reply:
That won’t work because it seems causing an infinite loop in optimizer. It gives me error messages like running out of max iterations.

Reply:
I see, it is due to predicate pushdown rule. I think we need general solution as @maropu said.

Reply:
Do you have a sample query in mind that produces `Project->Filter->Sample`? I've been trying to come up with a query that generates this plan.